### ISSUE TITLE
Call stack size exceeded/too much recursion: Using Angular 2

### ISSUE THREAD

**Issue by avidus3r** [DevExpress/testcafe#1481]
### Are you requesting a feature or reporting a bug?
Reporting a bug

### What is the current behavior?
When trying to execute click on a DOM element that has template bindings, I get the following errors (by browser)

Safari: TypeError: null is not an object (evaluating 'j.parseProxyUrl(e.responseURL).destUrl')
Firefox: Too much recursion
Chrome: Uncaught RangeError: Maximum call stack size exceeded

### What is the expected behavior?
Select an option within a select

### How would you reproduce the current behavior (if this is a bug)?
Run a test against a project in Angular 2 that interacts with a DOM element that has template bindings 

#### Provide the test code and the tested page URL (if applicable)

Tested page URL:

Test code

```features-and-specs.js
import { 
    Selector
} from 'testcafe';

export default class FeaturesAndSpecsPage {
    constructor () {
        const namespace = 'cd-specs';
        this.componentSelector = Selector(`${namespace}`);

        this.styleSelect        = Selector('#selectStyle');
        this.styleSelectOption  = this.styleSelect.find('option');
        this.trimsSelect        = Selector('#selectTrim');
        this.trimsSelectOption  = this.trimsSelect.find('option');
    }
}
```

```js
import FeaturesAndSpecs from './features-and-specs';
import { Selector } from 'testcafe';

let remoteUrl = 'http://local.caranddriver.com:3000';
let path = '/bmw/3-series/specs';

let testUrl = remoteUrl + path;
let subPage = testUrl + '/2017/3-series-Sedan/387943';

fixture `Features and Specs`
    .page `${testUrl}`;

// Page model
const page = new FeaturesAndSpecs();

test('Global page props', async t => {
    let componentSelector = await page.componentSelector;
    let expectedTitle = 'BMW 3-series';
    let pageTitle = Selector('page-title');
    await t
        .expect(componentSelector.exists).ok()
        .expect(pageTitle.innerText).contains(expectedTitle);
        
    let styleSelect = Selector('#selectStyle');
    let styleSelectOption = Selector('#selectStyle').find('option').withText('3-series Convertible');

    await t
        .expect(styleSelectOption.exists).ok()
        .click(styleSelect)
        .click(styleSelectOption) //<--- errors here
        .expect(styleSelect.value).eql('3-series Convertible');
});
```

### Specify your

* operating system: OSX 10.11.13
* testcafe version: 0.15.0
* node.js version: 6.9.2

**Comment by avidus3r** [comment]
It may be important to note that this project was not created with Angular 2 cli.
Also the project implements Angular Universal.

**Comment by miherlosev** [comment]
I see you provide a non-public url to reproduce this issue - http://local.caranddriver.com:3000
However, with http://www.caranddriver.com/bmw/3-series/specs/2017/3-series-Sedan/387943 I have reproduced another problem - https://github.com/DevExpress/testcafe-hammerhead/issues/1131.
We will fix it during the [next sprint] (https://github.com/DevExpress/testcafe-hammerhead/milestone/12)

We also want to be sure that the original problem is fixed.
Could you provide a public url to reproduce the problem described above?

**Comment by avidus3r** [comment]
Apologies, http://www.caranddriver.com is not completely on the Angular 2 stack. You can use http://stage-www.caranddriver.com/bmw/3-series/specs/2017/3-series-Sedan/387943

**Comment by miherlosev** [comment]
https://github.com/DevExpress/testcafe-hammerhead/issues/1132

**Comment by AlexanderMoskovkin** [comment]
fixed in `testcafe@0.16.0-alpha3`

**Comment by avidus3r** [comment]
Awesome, thank you.
